### PR TITLE
feat(progress bar): Gives constant execution feedback

### PR DESCRIPTION
**Pull request by macunha1**
Hurts a bit the performance but overall makes the process feels better,
since it gives constant feedback regarding the execution with metrics
about data being generated.

[asciicast](https://asciinema.org/a/vMXhQJcqtc2u64W2znG7KtW5Y)

## Changes out-of-scope
It was necessary to change the fixed version for dev packages due to sub dependencies version conflicts. More specifically `pycodestyle`;

<details>
  <summary>Click here to see the logs</summary>

```
There are incompatible versions in the resolved dependencies.
[pipenv.exceptions.ResolutionFailure]:   File "/usr/lib/python3.8/site-packages/pipenv/utils.py", line 726, in resolve_deps
[pipenv.exceptions.ResolutionFailure]:       req_dir=req_dir,
[pipenv.exceptions.ResolutionFailure]:   File "/usr/lib/python3.8/site-packages/pipenv/utils.py", line 480, in actually_resolve_deps
[pipenv.exceptions.ResolutionFailure]:       resolved_tree = resolver.resolve()
[pipenv.exceptions.ResolutionFailure]:   File "/usr/lib/python3.8/site-packages/pipenv/utils.py", line 395, in resolve
[pipenv.exceptions.ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]:       pipenv.exceptions.ResolutionFailure: ERROR: ERROR: Could not find a version that matches pycodestyle<2.6.0,>=2.5.0,>=2.6.0
[pipenv.exceptions.ResolutionFailure]:       Tried: 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.2.0, 2.2.0, 2.3.0, 2.3.0, 2.3.1, 2.3.1, 2.4.0, 2.4.0, 2.5.0, 2.5.0, 2.6.0, 2.6.0
[pipenv.exceptions.ResolutionFailure]:       Skipped pre-versions: 1.8.0.dev0, 1.8.0.dev0, 2.0.0a1, 2.0.0a1, 2.6.0a1, 2.6.0a1
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: ERROR: Could not find a version that matches pycodestyle<2.6.0,>=2.5.0,>=2.6.0
Tried: 2.0.0, 2.0.0, 2.1.0, 2.1.0, 2.2.0, 2.2.0, 2.3.0, 2.3.0, 2.3.1, 2.3.1, 2.4.0, 2.4.0, 2.5.0, 2.5.0, 2.6.0, 2.6.0
Skipped pre-versions: 1.8.0.dev0, 1.8.0.dev0, 2.0.0a1, 2.0.0a1, 2.6.0a1, 2.6.0a1
There are incompatible versions in the resolved dependencies.
```
<details>

### CODE REVIEW

**Comment by macunha1** [comment]
Performance benchmark results:

Without progress bar
```bash
> time ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json
ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json  22.54s user 0.46s system 101% cpu 22.761 total

> time ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json
ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json  22.10s user 0.46s system 101% cpu 22.256 total

> time ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json
ggenerator generate --spec /home/macunha1/Downloads/ggenerator/spec.json  22.11s user 0.55s system 101% cpu 22.364 total
```

With it
```bash
> time python /home/macunha1/Workspace/Datenworks/Public/ggenerator/main.py generate --spec spec.json
python /home/macunha1/Workspace/Datenworks/Public/ggenerator/main.py generate  22.94s user 0.56s system 101% cpu 23.241 total

> time python /home/macunha1/Workspace/Datenworks/Public/ggenerator/main.py generate --spec spec.json
python /home/macunha1/Workspace/Datenworks/Public/ggenerator/main.py generate  23.29s user 0.51s system 100% cpu 23.568 total
```

**Review by iRitiLopes — DISMISSED** [review]
I Like what you did. Excellent.

File: Pipfile [hunk 1]
```diff
@@ -4,14 +4,15 @@ url = "https://pypi.org/simple"
...
+flake8 = ">=3.7.9,<4.0.0"
+pytest-mock = ">=2.0.0,<3.0.0"
 autopep8 = "*"
-pytest-cov = "==2.8.1"
-moto = "==1.3.8"
+pytest-cov = ">=2.8.1,<3.0.0"
+moto = ">=1.3.8,<2.0.0"
 
 [packages]
+alive-progress = ">=1.5.1"
```
Comment (iRitiLopes) [inline Pipfile:15]: I see that you add a new dependency. Now we does not implemented something to update our setup.py to maintain updated the dependencies, can you add this lib on setup.py for that the installation of GGenerator deal with it and get it too?

**Review by iRitiLopes — COMMENTED** [review]
Exist some conflicts on Pipfile and Pipfile.lock too

File: Pipfile (hunk 1 above)
Comment (macunha1) [reply Pipfile:15]: @iRitiLopes sure, thanks for bringing that up

**Review by iRitiLopes — APPROVED** [review]
I tested here and is great. Even on windows haha. 
Is very useful to have some metrics like these.

**Pull request merged**